guppylang/compiler/expr_compiler.py

```diff
@@ -183,6 +188,50 @@ def visit_LocalCall(self, node: LocalCall) -> OutPortV:
         rets = [call.out_port(i) for i in range(len(type_to_row(func.ty.output)))]
         return self._pack_returns(rets, func.ty.output)
 
+    def visit_TensorCall(self, node: TensorCall) -> OutPortV:
+        func = self.visit(node.func)
+        args = [self.visit(arg) for arg in node.args]
+
+        assert isinstance(func.ty, TupleType)
+
+        rets: list[OutPortV] = []
+        remaining_args = args
+        for elem in self._unpack_tuple(func):
+            outs, remaining_args = self._compile_tensor_with_leftovers(
+                elem, remaining_args
+            )
+            rets.extend(outs)
+        assert remaining_args == []
+
+        if len(rets) == 1:
+            return rets[0]
+        else:
+            return self._pack_returns(rets, node.out_tys)
```

Comment:
Why is this `len == 1` special case needed?

Reply:
It should be redundant, duplicating the logic of `pack_returns`. I'll remove it